game/plugins/src/main/kotlin/gg/rsmod/plugins/content/skills/thieving/pickpocketing/pickpocket.kts

```diff
@@ -0,0 +1,69 @@
+package gg.rsmod.plugins.content.skills.thieving.pickpocketing
+
+import gg.rsmod.plugins.content.combat.isAttacking
+import gg.rsmod.plugins.content.combat.isBeingAttacked
+
+//for each npc in the enum
+PickpocketInfo.values().forEach { npcClass ->
+
+    //for each npc id in the current enum
+    npcClass.ids.forEach { npcID ->
+
+        //Extra check to ensure that the npc is pickpocket-able
+        if (world.definitions.get(NpcDef::class.java, npcID).options.contains("Pickpocket")) {
+
+            //Registers what to do with pickpocketing option
+            on_npc_option(npc = npcID, option = "pickpocket") {
+
+                //accesses the player queue
+                player.queue {
+                    if (!player.isAttacking() && !player.isBeingAttacked()) {
+                        if (player.inventory.hasSpace) {
+                            //checks if the player has the required level for the npc
+                            val thievLvl: Int = player.getSkills().getCurrentLevel(Skills.THIEVING)
+                            if (thievLvl >= npcClass.lvl) {
```

Comment:
Same for this one, just check if the thieving level is smaller and do a `continue;`